src/metpy/calc/thermo.py

```diff
@@ -955,6 +973,13 @@ def _parcel_profile_helper(pressure, temperature, dewpoint):
     temp_upper = moist_lapse(unique[::-1], temp_lower[-1]).to(temp_lower.units)
     temp_upper = temp_upper[::-1][indices]
 
+    # Do the virtual temperature correction for the parcel above the LCL
+    if assumptions.use_virtual_temperature:
+
+        # Calculate the mixing ratio and virtual temperature
+        rv_upper = mixing_ratio(saturation_vapor_pressure(temp_lcl), press_upper)
```

Comment:
The temperature being passed to the saturation_vapor_pressure() here should be the temperature *profile* at above the LCL (i.e. it should be "temp_upper"), *not* just the singular temperature at the LCL (that is, "temp_lcl"). In other words, the saturation vapor pressure of the parcel should decrease with height since it depends only on the parcel's temperature, which also decreases with height. The way it is written here assumes that the saturation vapor pressure at *and above* the LCL will be constant and equal to its value at the LCL (since you are passing "temp_lcl" to the saturation_vapor_pressure() function), which is incorrect. Therefore, the computed mixing ratio of the parcel at/above the LCL and the final virtual temperature profile of the parcel will also be incorrect.

Otherwise, the rest of the calculations appear to be correct as far as I can tell (with the caveat that the LCL that is being used here should be that computed from the *uncorrected* parcel and environmental temperature profiles, as discussed in #1735).

Reply:
@aschueth and @sgdecker who commented in #1735 and #461, respectively, may want to chime in as well.